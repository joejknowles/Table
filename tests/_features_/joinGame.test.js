import createBrowser from './setup/phantom';
import createAppStarter from './setup/server';
import res from '../../client/src/resources/pages/startScreen';

let browser;
let appStarter;
let host;

beforeAll(async () => appStarter = await createAppStarter(5050));

beforeEach(async () =>{
  host = appStarter();
  browser = await createBrowser(host.port);
});

it('clicking "join as player" button shows a deck of cards', async () => {
  await browser.visit('/');
  const playerJoinButton = await browser.click('.player-join-button');
  const cardDeck = await browser.find('.play-card');
  expect(cardDeck.className).toBe('play-card btn');
});

it('clicking "join as player" button shows a deck of cards', async () => {
  await browser.visit('/');
  const playerJoinButton = await browser.click('.table-join-button');
  const cardDeck = await browser.find('.pile');
  expect(cardDeck.className).toBe('pile');
});

afterEach(() => {
  browser.exit();
  host.server.close();
});
