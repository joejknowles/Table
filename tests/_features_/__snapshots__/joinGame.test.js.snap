exports[`test join game page matches snapshot 1`] = `"<!DOCTYPE html><html lang=\"en\"><head><meta charset=\"utf-8\"><meta name=\"viewport\" content=\"width=device-width,initial-scale=1\"><link rel=\"shortcut icon\" href=\"/favicon.ico\"><title>React App</title><link href=\"/static/css/main.90dce555.css\" rel=\"stylesheet\"></head><body><div id=\"root\"><div data-reactroot=\"\" class=\"App\"><button class=\"player-join-button\">Join as player</button><button class=\"table-join-button\">Join as table</button></div></div><script type=\"text/javascript\" src=\"/static/js/main.83edb080.js\"></script></body></html>"`;
